src/autocat/learning/sequential.py

```diff
@@ -172,50 +168,379 @@ def update(self, structures: List[Atoms], labels: Array):
+
+        uncertainties:
+            Uncertainties for all structures in the DesignSpace
+
+        Returns
+        -------
+
+        parent_idx:
+            Index/indices of the selected candidates
+
+        max_scores:
+            Maximum scores (corresponding to the selected candidates)
+
+        aq_scores:
+            Calculated scores using `acquisition_function` for the entire DesignSpace
+        """
+        ds_size = len(design_space)
+
+        if allowed_idx is None:
+            if True in np.isnan(design_space.design_space_labels):
+                allowed_idx = np.where(np.isnan(design_space.design_space_labels))[0]
+            else:
+                allowed_idx = np.ones(ds_size, dtype=bool)
+
+        hhi_scores = np.ones(ds_size)
+        if self.include_hhi:
+            hhi_scores = calculate_hhi_scores(
+                design_space.design_space_structures, self.hhi_type
+            )
+
+        segreg_energy_scores = np.ones(ds_size)
+        if self.include_segregation_energies:
+            segreg_energy_scores = calculate_segregation_energy_scores(
+                design_space.design_space_structures
+            )
+
+        aq = self.acquisition_function
+        if aq == "Random":
+            aq_scores = (
+                np.random.choice(ds_size, size=ds_size, replace=False)
+                * hhi_scores
+                * segreg_energy_scores
+            )
+
+        elif aq == "MU":
+            if uncertainties is None:
+                msg = "For 'MU', the uncertainties must be supplied"
+                raise CandidateSelectorError(msg)
+            aq_scores = uncertainties.copy() * hhi_scores * segreg_energy_scores
+
+        elif aq == "MLI":
+            if uncertainties is None or predictions is None:
+                msg = "For 'MLI', both uncertainties and predictions must be supplied"
+                raise CandidateSelectorError(msg)
+            target_window = self.target_window
+            aq_scores = (
+                np.array(
+                    [
+                        get_overlap_score(
+                            mean, std, x2=target_window[1], x1=target_window[0]
+                        )
+                        for mean, std in zip(predictions, uncertainties)
+                    ]
+                )
+                * hhi_scores
+                * segreg_energy_scores
+            )
+
+        num_candidates_to_pick = self.num_candidates_to_pick
+        if num_candidates_to_pick == 1:
+            next_idx = np.array([np.argmax(aq_scores[allowed_idx])])
+            max_scores = [np.max(aq_scores[allowed_idx])]
+
+        else:
+            next_idx = np.argsort(aq_scores[allowed_idx])[-num_candidates_to_pick:]
+            sorted_array = aq_scores[allowed_idx][next_idx]
+            max_scores = list(sorted_array[-num_candidates_to_pick:])
+        parent_idx = np.arange(aq_scores.shape[0])[allowed_idx][next_idx]
```

Comment:
I think we can avoid this `if-else` block, no? The code/logic in the `else` block should be generalizable to `num_candidates_to_pick = 1` as well, right?